Maven failsafe plugin skipping some ITs

**Describe the bug**
The current version of the Maven failsafe plugin is `3.0.0-M4`. This version is not correctly running ITs, some are being skipped. This doesn't happen with other versions of the plugin (e.g. `3.0.0-M5` and `2.22.2`). The [fixed bugs for the subsequent version of the failsafe plugin](https://issues.apache.org/jira/browse/SUREFIRE-1570?jql=project%20%3D%20SUREFIRE%20AND%20issuetype%20%3D%20Bug%20AND%20fixVersion%20%3D%203.0.0-M5%20ORDER%20BY%20priority%20DESC%2C%20key) don't seem to mention a bug that would have caused this behaviour. 

It is also possible to get version `3.0.0-M4` to run these ITs, which is happening in PR #2841, but that PR has no changes to the tests or plugin config and so the cause of inconsistent skipping with this version is unclear.

Upgrading the plugin version to the latest should fix this. Although it might be good to have an extra checks for test classes with apparent 0 tests run (this is expected only for classes which are suites).

**To Reproduce**
On v2-alpha ([this commit at the time of writing](https://github.com/gchq/Gaffer/tree/803a3747379e8b1abf19df8b97a7e23bd0d592fb)), run `mvn clean install -Pcoverage -pl :proxy-store`. At the end of the failsafe IT report you can see the IT class `ProxyStoreResponseDeserialiserIT` has `Tests run: 0`, but this class contains one test which should be run.

Changing the plugin version and rerunning the above causes more tests to be run. In the case of `ProxyStoreResponseDeserialiserIT` the IT fails. This failure is currently hidden because the IT is not being run. See additional context.

**Expected behavior**
No classes being reported as having `Tests run: 0`, unless they actually contain no test methods (as with `@Suite` classes).

**Stack trace and errors**
With the current version of the plugin:
```
[INFO] Running uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreBasicIT
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.062 s - in uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreBasicIT
[INFO] Running uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT
[INFO] Tests run: 0, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.042 s - in uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT
[INFO] 
[INFO] Results:
[INFO] 
[WARNING] Tests run: 151, Failures: 0, Errors: 0, Skipped: 15
```

With version of the plugin which work as expected:
```
[INFO] Running uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreBasicIT
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.062 s <<< FAILURE! - in uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreBasicIT
[INFO] Running uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.049 s <<< FAILURE! - in uk.gov.gchq.gaffer.proxystore.integration.ProxyStoreResponseDeserialiserIT
[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   ProxyStoreBasicIT.beforeAll:108 » Processing Failed to start Grizzly HTTP serv...
[ERROR]   ProxyStoreResponseDeserialiserIT.beforeAll:61 » Processing Failed to start Gri...
[INFO] 
[ERROR] Tests run: 153, Failures: 0, Errors: 2, Skipped: 15
```

**Additional context**
The error which this issue has uncovered is being tracked in #2849 which provides further context.

--- a/pom.xml
+++ b/pom.xml
@@ -54,7 +54,7 @@
         <junit-suite.version>1.9.0</junit-suite.version>
         <assertj.version>3.23.1</assertj.version>
         <mockito.version>4.6.1</mockito.version>
-        <failsafe.version>3.0.0-M4</failsafe.version>
+        <failsafe.version>3.0.0-M7</failsafe.version>
         <slf4j.api.version>1.7.36</slf4j.api.version>
 
         <scala.minor.version>2.11</scala.minor.version>

--- a/pom.xml
+++ b/pom.xml
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <!--
-  ~ Copyright 2016-2022 Crown Copyright
+  ~ Copyright 2016-2023 Crown Copyright
   ~
   ~ Licensed under the Apache License, Version 2.0 (the "License");
   ~ you may not use this file except in compliance with the License.

